Sources/Basic/FileSystem.swift

```diff
@@ -269,6 +272,16 @@ private class LocalFileSystem: FileSystem {
         let cwdStr = FileManager.default.currentDirectoryPath
         return try? AbsolutePath(validating: cwdStr)
     }
+    
+    var homeDirectory: AbsolutePath? {
+        let homeDirStr: String
+        if #available(OSX 10.12, *) {
+            homeDirStr = FileManager.default.homeDirectoryForCurrentUser.path
+        } else {
+            homeDirStr = NSHomeDirectory()
+        }
```

Comment:
@giginet did you test this condition on Linux?

Reply:
Not yet 😞 

However, I checked this implementation may work on Linux.

https://github.com/apple/swift-corelibs-foundation/blob/5b3dae9518684ef1bbceb47221f09e75af1fcef8/Foundation/NSPathUtilities.swift#L568